### ISSUE TITLE
More generic versions of concrete types

### ISSUE THREAD

**Issue by max-heller** [facebookincubator/smt2utils#36]
Would it be possible to add generic versions of the following concrete types to `smt2parser::visitors` similar to how `Identifier`, `Index`, and the `*Dec` types are defined?

- `SExpr` (generic over `Constant`, `Symbol`, `Keyword`)
- `QualIdentifier` (generic over `Identifier`, `Sort`)
- `Command` (generic over `Symbol`, `Sort`, `Keyword`, `Term`)

**Comment by matbd** [comment]
@max-heller Sorry, I just saw your question. Are you looking for definitions like `smt2parser::concrete::{SExpr, QualIdentifier, Command}` (basically the enums in https://docs.rs/smt2parser/0.6.0/smt2parser/concrete/index.html) but where elementary types are still generic?

**Comment by matbd** [comment]
I didn't know about default type values for containers. This is nice!

So, I suppose every struct/enum in `concrete` would look like this:
```
pub enum SExpr<Constant = Constant, Symbol = Symbol, Keyword = Keyword> {
    Constant(Constant),
    Symbol(Symbol),
    Keyword(Keyword),
    Application(Vec<Self>),
}
```

It looks like the trait `Display` should be easy enough to generalize. Not sure about the rest of the `concrete` module (but we could try).

**Comment by ma2bd** [comment]
@max-heller Thanks again for the PR. I just released smt2parser v0.6.1
